src/transport/SecureSession.h

```diff
@@ -209,17 +221,45 @@ class SecureSession : public Session
     SessionMessageCounter & GetSessionMessageCounter() { return mSessionMessageCounter; }
 
 private:
-    Type mSecureSessionType;
+    enum class State : uint8_t
+    {
+        // kPending denotes a secure session object that is internally
+        // reserved by the stack before and during session establishment.
+        //
+        // Although the stack can tolerate eviction of these (releasing one
+        // out from under the holder would exhibit as CHIP_ERROR_INCORRECT_STATE
+        // during CASE or PASE), intent is that we should not and would leave
+        // these untouched until CASE or PASE complete.
+        //
+        // During this stage, the reference counter is hold by the PairingSession
+        kPairing = 1,
+
+        // The session is active, ready for use. During this stage, the
+        // reference counter increased by 1 in Activate, and will be decreased
+        // by 1 when MarkForRemoval is called.
+        kActive = 2,
+
+        // The session is pending for removal, all SessionHolders are already
+        // cleared during MarkForRemoval, no future SessionHolder is able grab
+        // this session, when all SessionHandles goes out of scope, the session
+        // object will be released automatically.
+        kPendingRemoval = 3,
+    };
+
+    friend class SecureSessionDeleter;
+    SecureSessionTable & mTable;
+    State mState;
+    const Type mSecureSessionType;
+    NodeId mLocalNodeId = kUndefinedNodeId;
+    NodeId mPeerNodeId  = kUndefinedNodeId;
+    CATValues mPeerCATs = CATValues{};
     const uint16_t mLocalSessionId;
-    NodeId mLocalNodeId;
-    NodeId mPeerNodeId;
-    CATValues mPeerCATs;
-    uint16_t mPeerSessionId;
+    uint16_t mPeerSessionId = 0;
 
     PeerAddress mPeerAddress;
-    System::Clock::Timestamp mLastActivityTime;     ///< Timestamp of last tx or rx
-    System::Clock::Timestamp mLastPeerActivityTime; ///< Timestamp of last rx
-    ReliableMessageProtocolConfig mMRPConfig;
+    System::Clock::Timestamp mLastActivityTime     = System::SystemClock().GetMonotonicTimestamp(); ///< Timestamp of last tx or rx
+    System::Clock::Timestamp mLastPeerActivityTime = System::SystemClock().GetMonotonicTimestamp(); ///< Timestamp of last rx
```

Comment:
This does not seem right at for a session that we pre-allocate for a handshake and which might never have had an rx.  Why was this change made?

Reply:
This change is just a refactor, moved from the old constructor.

We can address this issue separately.

Reply:
Ah, I see, the non-test constructor called the one that did this.  OK.